packages/react/src/select/trigger/SelectTrigger.tsx

```diff
@@ -160,6 +160,10 @@ export const SelectTrigger = React.forwardRef(function SelectTrigger(
         store.set('touchModality', pointerType === 'touch');
       },
       onKeyDown(event) {
+        if (disabled || readOnly) {
+          return;
+        }
+
         keyboardActiveRef.current = true;
 
         if (event.key === 'ArrowDown') {
```

Comment:
Should this code block actually exist - is there a test covering it? `useListNavigation` should handle the logic already

Reply:
That makes sense. It's clear now that the logic in `onKeyDown` is redundant since it's covered by `useListNavigation`.

I have one follow-up question, though. Am I correct in thinking that the `keyboardActiveRef` is still necessary to track keyboard activity?